ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -298,21 +298,26 @@ Joiner getJoiner(final DataSource.DataSourceType leftType,
           .getKeyField()
           .resolve(schemaKStream.getSchema(), builder.getKsqlConfig());
 
-      if (keyField.isPresent()
-          && !keyFieldName.equals(SchemaUtil.ROWKEY_NAME)
-          && !SchemaUtil.matchFieldName(keyField.get(), expectedKeyField)) {
+      final boolean namesMatch = keyField
+          .map(field -> SchemaUtil.matchFieldName(field, expectedKeyField))
+          .orElse(false);
+
+      if (namesMatch || keyFieldName.equals(SchemaUtil.ROWKEY_NAME)) {
+        return (SchemaKTable) schemaKStream;
+      }
+
+      if (!keyField.isPresent()) {
```

Comment:
nit: move this above the `namesMatch` and then just use `keyField.get()`  instead of `map`

Reply:
That would break the code as its valid for `keyField` to be `empty` if the new key field is `ROWTIME`.